### PR TITLE
4/Create-List-of-Ingredients

### PR DESCRIPTION
**Pull request by omorgenroth**
main content:
- Added Ingredient Category to Backend/DB (php)
- Created own Serializer for DishEntity to beautify my the JSON Object in the Response of my API (php) 
- Created a new Page and Component to show all Ingredients from selected Dishes ( w/o duplicates) separated by categories. (js)

### CODE REVIEW

File: backend/src/Repository/IngredientCategoryRepository.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ */
+class IngredientCategoryRepository extends ServiceEntityRepository
+{
+    public function __construct(ManagerRegistry $registry)
+    {
+        parent::__construct($registry, IngredientCategory::class);
+    }
+
+    // /**
+    //  * @return IngredientCategory[] Returns an array of IngredientCategory objects
```
Comment (tine-krueger) [inline backend/src/Repository/IngredientCategoryRepository.php:23]: delete this comment, please

File: backend/src/Serializer/DishSerializer.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            foreach($elements as $element) {
+                $this->setArray($element);
+            }
+        } else {
+            $this->setArray($elements);
+        }
+        
+        return \json_encode($this->elementAsArray);
+    }
+
```
Comment (tine-krueger) [inline backend/src/Serializer/DishSerializer.php:52]: 👍

File: frontend/src/App.js [hunk 3]
```diff
@@ -26,33 +30,74 @@ function App() {
...
+          newList = [...newList, existingItem]
+          list = newList
+        } else {
+          list.push(currentElement)
+        }
+        return list
+      },
+      []
+    )
+
```
Comment (tine-krueger) [inline frontend/src/App.js:97]: Is this what Andreas wanted you to use the backend?

File: frontend/src/Pages/DishesSelectedPage.js [hunk 4]
```diff
@@ -1,25 +1,35 @@
 import PropTypes from 'prop-types'
 import styled from 'styled-components/macro'
 import HeaderOverlay from '../Components/HeaderOverlay/HeaderOverlay'
-import ListItem from '../Components/ListItem/ListItem'
+import DishListItem from '../Components/DishListItem/DishListItem'
 
 SelectedDishesPage.propTypes = {
   dishes: PropTypes.array,
   onDeleteItem: PropTypes.func,
 }
```
Comment (tine-krueger) [inline frontend/src/Pages/DishesSelectedPage.js:9]: are they required? in this case you can use 
```suggestion
onDeleteItem: PropTypes.func.isRequired
```
maybe better for error-handling

File: frontend/src/App.js [hunk 5]
```diff
@@ -26,33 +30,74 @@ function App() {
...
+        <Route exact path="/">
+          <LandingPage loading={isLoading} />
+        </Route>
+        <Route path="/home">
+          <HomePage />
+        </Route>
+        <Route path="/dishes/selected">
+          <DishesSelectedPage
+            selectedDishes={selectedDishes}
+            onDeleteItem={(newSelectedDishes) =>
```
Comment (janabremer) [inline frontend/src/App.js:43]: the naming "newSelectedDishes" is a bit contradictory together with delete... maybe updatedSelectedDishes?

File: frontend/src/App.js [hunk 6]
```diff
@@ -26,33 +30,74 @@ function App() {
...
+  function createIngredientList() {
+    let flatIngredients = []
+    selectedDishes.forEach((dish) => {
+      let i
+      for (i = 0; i < dish.ingredients.length; i++) {
+        flatIngredients.push(dish.ingredients[i])
+      }
+    })
+
+    const reducedIngredientList = flatIngredients.reduce(
```
Comment (janabremer) [inline frontend/src/App.js:81]: 👍

File: frontend/src/Components/Accordion/Accordion.css [hunk 7]
```diff
@@ -0,0 +1,60 @@
+@import url('https://fonts.googleapis.com/css?family=Open+Sans:400,600&display=swap');
```
Comment (janabremer) [inline frontend/src/Components/Accordion/Accordion.css:1]: why do you use css here? the styling method should be consistent.
Besides: don't forget to delete the comments.

File: frontend/src/Components/Accordion/Accordion.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
+import { useState, useRef } from 'react'
```
Comment (janabremer) [inline frontend/src/Components/Accordion/Accordion.js:1]: I am curious how your accordion looks like :)

File: frontend/src/Components/IngredientListItem/IngredientListItem.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+  props: PropTypes.object,
+  onClick: PropTypes.func,
+}
+
+export default function IngredientListItem({ props, onClick }) {
+  return (
+    <ItemWrapper onClick={() => onClick(props.id)} checked={props.isSelected}>
+      <IconWrapper>
+        {props.isSelected ? (
+          <CheckedIcon data-testid="checked" />
```
Comment (janabremer) [inline frontend/src/Components/IngredientListItem/IngredientListItem.js:15]: didn't know this one 👌

**Review by janabremer — APPROVED** [review]
Looks good to me! I only had some small comments.

File: backend/src/Repository/IngredientCategoryRepository.php (hunk 1 above)
Comment (omorgenroth) [reply backend/src/Repository/IngredientCategoryRepository.php]: Done, good point!

File: frontend/src/App.js (hunk 3 above)
Comment (omorgenroth) [reply frontend/src/App.js:95]: No a bit of transforming on client side needs to be done anyway to fit the needs of my app.

File: frontend/src/App.js (hunk 5 above)
Comment (omorgenroth) [reply frontend/src/App.js]: Did that, thanks for the hint.

**Pull request merged**